kphp-inspector/src/file_searcher.js

```diff
@@ -38,42 +39,139 @@ function doesFileContainFunction(fullFileName) {
-  let qParts = splitUserSearchStr(q);
-  if (!qParts.length) {
-    return [];
+function performSearchForFunction(query) {
+  const strict = query.startsWith('\\');
+  utils.debug(`Start search function by query: ${query}, strict: ${strict}`);
+
+  if (strict) {
+    // Convert the name to the form that KPHP uses for files.
+    const filename = convertFqnToCppFileName(query);
+    // When searching for such a string, when we find it, we will be sure
+    // that this is the required file, and not a function file with the same
+    // suffix or prefix, for example:
+    //   filename    = "someFunc"
+    //   found_files = ["src/someFunc.cpp", "src/someFuncOther.cpp"]
+    const filename_with_slash_and_dot = "/" + filename + '\\.';
+    utils.debug(`Filename for search: ${filename_with_slash_and_dot}`);
+
+    const files = findFunctionFilesWithNameLike(filename_with_slash_and_dot, false);
+    utils.debug(`Found files: ${files.length === 0 ? 'nothing' : files.join(', ')}`);
+
+    return files.filter(file => doesFileContainFunction(file));
   }
-  let longestQLen = Math.max(...qParts.map(p => p.length));
-  let longestQ = qParts.find(p => p.length === longestQLen);
 
-  /** @var {string[]} */
-  let output = child_process.execSync(`find ${env.RUN_ARGV.KPHP_COMPILED_ROOT} -iname "*${longestQ}*"`).toString().split('\n');
-  // strip out duplicates and strange files: leave only .cpp and .h, also filter out .h files of classes
-  output = output.filter(fn =>
-    fn.endsWith('.cpp') || (fn.endsWith('.h') && !output.includes(fn.substr(0, fn.length - 2) + '.cpp') && !fn.includes('cl/C@')));
+  let [qParts, longestQ] = queryToParts(query);
+  const files = findFunctionFilesWithNameLike(longestQ);
+  utils.debug(`Found files: ${files.length === 0 ? 'nothing' : files.join(', ')}`);
 
-  return output.filter(fn => doesFileSatisfyQ(fn, qParts) && doesFileContainFunction(fn));
+  return files.filter(fn => doesFileSatisfyQ(fn, qParts) && doesFileContainFunction(fn));
 }
 
 /**
- * @param {string} q Search string from console
+ * @param {string} query Search string from console
  * @return {string[]} Full file names, to be passed to parseCppClass()
  */
-function performSearchForClass(q) {
-  let qParts = splitUserSearchStr(q);
-  if (!qParts.length) {
-    return [];
+function performSearchForClass(query) {
+  const strict = query.startsWith('\\');
+  utils.debug(`Start search class by query: ${query}, strict: ${strict}`);
+
+  if (strict) {
+    // Convert the name to the form that KPHP uses for files.
+    const className = convertFqnToCppFileName(query);
+    const classnameWithDot = className + '\\.';
+    utils.debug(`Filename for search: ${classnameWithDot}`);
+
+    const files = findFilesWithNameLike(classnameWithDot);
+    utils.debug(`Found files: ${files.length === 0 ? 'nothing' : files.join(', ')}`);
+
+    return files.filter(file => file.endsWith('.h'));
+  }
+
+  let { qParts, longestQ } = queryToParts(query);
```

Comment:
`[qParts, longestQ]` is above, but here you use `{}` destructuring.
I'd suggest to provide the `queryToParts` a better typing than it has now.

Reply:
Fixed.